### ISSUE TITLE
HTTP Action reporting success even though an "Unexpected Error" occurred

### ISSUE THREAD

**Issue by arnoventer** [Azure/logicapps#306]
The HTTP action of the logic app fails, however the status is reported is success.
I expect the Status to report "Failure" or "Error" or something.
I have only picked this up with a CORS issue, and haven't tested if the same behaviour happens with other errors.

![Screenshot 2021-06-07 at 17 50 54](https://user-images.githubusercontent.com/4471638/121050766-26470d80-c7b9-11eb-9bcb-c42a2f636408.png)
![Screenshot 2021-06-07 at 17 51 13](https://user-images.githubusercontent.com/4471638/121050774-2810d100-c7b9-11eb-8f4a-a57f95218cc2.png)

**Comment by HarisAbdulkareem** [comment]
@arnoventer : Are you still getting this error? Were you able to verify if the calls are actually reaching the HTTP endpoint, while you get this "unexpected error"?

**Comment by arnoventer** [comment]
hi @HarisAbdulkareem , sorry man, I could have sworn I closed this. We managed to identify the cause: the app service plan that the logic apps were configured on had a vnet and private network configured. So what was happening is that the jobs were succeeding, but when I didn't have my VPN switched on (to get onto the same network as the vnet and private network), I wasn't able to view the _actual_ results, and that was were the CORS issue was coming from (as it couldn't actually connect to that network to retrieve the results). I hope that makes sense for future googlers.
